tests/check_check_master.c

```diff
@@ -455,23 +470,61 @@ START_TEST(test_check_failure_msgs)
     tr = tr_fail_array[i - passed];
     ck_assert_msg(tr != NULL, NULL);
     got_msg = tr_msg(tr);
-    expected_msg = master_tests[i].msg;
+    expected_msg = master_test->msg;
+
+    unsigned char not_equal = 0;
+
+    switch (master_test->msg_type) {
+    case CK_MSG_TEXT:
+      if (strcmp(got_msg, expected_msg) != 0) {
+        not_equal = 1;
+      }
+      break;
+#if ENABLE_REGEX
+    case CK_MSG_REGEXP: {
+      regex_t re;
+      fprintf(stderr, "regex: %s %s\n", expected_msg, got_msg);
+      int reg_err = regcomp(&re, expected_msg, REG_EXTENDED | REG_NOSUB);
+      if (reg_err) {
+        char err_text[256];
+        regerror(reg_err, &re, err_text, sizeof(err_text));
+        ck_assert_msg(reg_err == 0, "regcomp: %s", err_text);
+      }
+      reg_err = regexec(&re, got_msg, (size_t) 0, NULL, 0);
+      regfree(&re);
+      if (reg_err) {
+        not_equal = 1;
+      }
+      break;
+    }
+#endif /* ENABLE_REGEX */
+    }
+    
+    if (not_equal) {
+      const char *msg_type_str;
+      switch(master_test->msg_type) {
+#if ENABLE_REGEX
+      case CK_MSG_REGEXP:
+        msg_type_str = " regexp";
+        break;
+#endif
+      default:
+        msg_type_str = "";
+      }
 
-    if (strcmp(got_msg, expected_msg) != 0) {      
-      char *emsg;
-      char *tmp = (char *)malloc(MAXSTR);
-      snprintf(tmp, MAXSTR,"For test %d:%s:%s Expected %s, got %s",
-               i, master_tests[i].tcname, master_tests[i].test_name,
-			   expected_msg, got_msg);
+      char _emsg[MAXSTR];
```

Comment:
small thing, but can you rename _emsg to something else? A leading '_' implies that it is internal or something else, when really you mean 'non escaped string'.